conmon-rs/server/src/child_reaper.rs

```diff
@@ -158,7 +158,7 @@ impl ChildReaper {
         grandchild_pid: u32,
     ) -> Result<()> {
         let mut map = lock!(locked_grandchildren);
-        map.retain(|_, v| v.pid == grandchild_pid);
+        map.retain(|_, v| v.pid != grandchild_pid);
         Ok(())
```

Comment:
Hm, I think this is the right change, we can observe CRI-O CI after this merge to double check if it breaks anything.

Can we further simplify this please?

Something like this should be fine:
```rust
        Ok(lock!(locked_grandchildren).retain(|_, v| v.pid == grandchild_pid))
```

Reply:
I would prefer an explicit `Ok(())` statement, but I'm ok with both styles:
```rs
lock!(locked_grandchildren).retain(|_, v| v.pid != grandchild_pid);
Ok(())
```

What do you think?

Reply:
I'm not opposed to any of the above, thank you!